### PR TITLE
multi filters and faceted autocomplete component for FastFT

### PR DESCRIPTION
**Pull request by i-like-robots**
:open_mouth: 

facets

boom

(inside this PR there is some nice code and there is some idiosyncratic code that I think needs [refactoring](https://github.com/Financial-Times/next-stream-page/issues/922)).

### CODE REVIEW

File: shared/components/stream-filters/_tag-autocomplete.scss [hunk 1]
```diff
@@ -0,0 +1,136 @@
+// scss-lint:disable all
+.ReactTags {
```
Comment (ironsidevsquincy) [inline shared/components/stream-filters/_tag-autocomplete.scss]: odd class name?

File: shared/components/stream-filters/index.js [hunk 2]
```diff
@@ -1,65 +1,184 @@
...
 		};
 	},
 
-	handleClick(e) {
-		e.preventDefault();
+	componentDidUpdate(oldProps, oldState) {
+		if (this.state.tags.length !== oldState.tags.length) {
+
+			const tagIds = this.state.tags.map(tag => tag.id);
+			window.history.pushState(null, null, `?filterBy=${tagIds.join()}`);
```
Comment (ironsidevsquincy) [inline shared/components/stream-filters/index.js]: how's this behave on IE9?

File: shared/components/stream-filters/index.js [hunk 3]
```diff
@@ -1,65 +1,184 @@
...
 
-		this.setState({ loading: true });
+		const query = [
+			`tagged=${tagIds.join()}`,
+			`partial=${input}`,
+			'minimum=0',
+			'count=6'
+		];
+
+		return jsonp(`https://tag-facets-api.ft.com/?${query.join('&')}`)
```
Comment (ironsidevsquincy) [inline shared/components/stream-filters/index.js]: can you use standard fetch for browsers that understand cors?

File: test/shared/components/stream-filters.test.js [hunk 4]
```diff
@@ -42,17 +46,17 @@ function teardownInstance() {
...
 
-		wait();
-	});
-}
+// 		wait();
+// 	});
+// }
 
-describe('Stream filters component', () => {
+describe.only('Stream filters component', () => {
```
Comment (ironsidevsquincy) [inline test/shared/components/stream-filters.test.js]: :bomb:

File: test/shared/components/stream-filters.test.js [hunk 5]
```diff
@@ -42,17 +46,17 @@ function teardownInstance() {
...
+// 		function wait() {
+// 			setTimeout(() => (cb() ? resolve : wait)(), 10);
+// 		}
 
-		wait();
-	});
-}
+// 		wait();
+// 	});
+// }
```
Comment (ironsidevsquincy) [inline test/shared/components/stream-filters.test.js]: needed?

File: shared/components/stream-filters/_tag-autocomplete.scss (hunk 1 above)
Comment (i-like-robots) [inline shared/components/stream-filters/_tag-autocomplete.scss]: external component

File: shared/components/stream-filters/index.js (hunk 2 above)
Comment (i-like-robots) [inline shared/components/stream-filters/index.js]: Ooh... I guess it doesn't. Not ideal. Will check stats to see if it's worth doing anything.

**Comment by andygnewman** [comment]
:+1:

File: shared/components/stream-filters/index.js (hunk 3 above)
Comment (i-like-robots) [inline shared/components/stream-filters/index.js]: had a look into this, didn't look like a simple fix... IE9 does support CORS, kind of but it's the same =S

**Pull request merged**